src/algebra/continued_fractions/definitions/basic.lean

```diff
@@ -0,0 +1,321 @@
+section coe
+/-- Fix another type `β` and assume `α` can be converted to `β`. -/
+variables {β : Type*} [has_coe α β]
+
+/-- Coerce a sequence by elementwise coercion. -/
+def seq.coe_to_seq : has_coe (seq α) (seq β) := ⟨seq.map (λ a, (a : β))⟩
+
+local attribute [instance] seq.coe_to_seq
+
+/-- Coerce a gcf by elementwise coercion. -/
+instance has_coe_to_generalized_continued_fraction : has_coe (gcf α) (gcf β) :=
+⟨λ ⟨h, s⟩, ⟨(h : β), (s : seq $ gcf.pair β)⟩⟩
+
+@[simp, move_cast]
+lemma coe_to_generalized_continued_fraction {g : gcf α} :
+(↑(g : gcf α) : gcf β) = ⟨(g.h : β), (g.s : seq $ gcf.pair β)⟩ :=
+by { cases g, refl }
+
+end coe
+end generalized_continued_fraction
+
+/-
+We now continue to define simple continued fractions. A *simple continued fraction* (scf) is a gcf
+whose partial numerators are all equal to one.
+
+                                1
+                h + --------------------------
+                                  1
+                      b₀ + --------------------
+                                    1
+                            b₁ + --------------
+                                        1
+                                  b₂ + --------
+                                      b₃ + ...
+
+
+Again, for convenience, one writes `[h; b₀, b₁, b₂,...]`.
+-/
+
+/--
+A generalized continued fraction is a simple continued fraction (scf) if all partial numerators are
+equal to one.
+-/
+def generalized_continued_fraction.is_simple_continued_fraction
+(g : generalized_continued_fraction α) [has_one α] :
+  Prop :=
+∀ (n : ℕ) (aₙ : α), g.partial_numerators.nth n = some aₙ → aₙ = 1
+
+variable (α)
+/--
+A simple continued fraction (scf) is a generalized continued fraction  whose partial numerators are
+equal to one. It is the subtype of gcfs that satisfy
+`generalized_continued_fraction.is_simple_continued_fraction`.
+ -/
+def simple_continued_fraction [has_one α] :=
+{g : generalized_continued_fraction α // g.is_simple_continued_fraction}
+
+variable {α}
+/- Interlude: define some expected coercions. -/
+namespace simple_continued_fraction
+open generalized_continued_fraction as gcf
+open simple_continued_fraction as scf
+variable [has_one α]
+
+/-- Lift a scf to a gcf using the inclusion map. -/
+instance has_coe_to_generalized_continued_fraction : has_coe (scf α) (gcf α) :=
+by {unfold scf, apply_instance}
+
+@[simp, elim_cast]
+lemma coe_to_generalized_continued_fraction {s : scf α} : (↑s : gcf α) = s.val := rfl
+
+end simple_continued_fraction
+
+/-
+Next, we define (regular) continued fractions. A *(regular) continued fraction* ((r)cf) is a simple
+continued fraction whose partial denominators `bᵢ` are all positive, i.e. `0 < bᵢ`.
+-/
+
+/--
+A simple continued fraction is a (regular) continued fraction ((r)cf) if all partial denominators
+are positive.
+-/
+def simple_continued_fraction.is_regular_continued_fraction [has_one α] [has_zero α] [has_lt α]
+(s : simple_continued_fraction α) :
```

Comment:
Same comment: indent and merge with the next line.

Reply:
Fixed.